stdlib/runtime/numbers.gr

```diff
@@ -1245,21 +1296,33 @@ export let coerceNumberToInt32 = (x: Number) => {
     // can possibly fail
     newInt32(coerceNumberToWasmI32(WasmI32.toGrain(x) : Number))
   }
-  WasmI32.toGrain(result): Int32
+  let ret = WasmI32.toGrain(result): Int32
+  if (WasmI32.fromGrain(ret) != WasmI32.fromGrain(x)) {
```

Comment:
This feels like a code smell. Shouldn't a function incRef the value it's returning (even if it's an argument)?

Reply:
I believe this is correct as written. If we call `coerceNumberToInt32(x)`, we expect to receive back `x` with a net refcount increase of +1 (excluding the `incRef` we do as part of the function call):
```
caller context: x <refcount: n>
function call: incRef(x) -> <refcount: n+1>
~~ in coerceNumberToInt32 ~~
let result = x <refcount: n+1> // simplified; this is the only case where this `if` becomes relevant
let ret = WasmI32.toGrain(result): Int32 == x <refcount: n+1>
```
at this point, we want to return `ret` as-is, with no decrease in refcount.

In contrast, if we call with a non-Int32 value, then we _do_ want to decRef here, since `result` is a different object.

Reply:
My bad! I guess I shouldn't review PRs late—I thought this was a consuming function of `coerceNumberToInt32` that was compensating for incorrect behavior. This _IS_ `coerceNumberToInt32`, just not being incorrect 😂  Ignore me, carry on 🙈